File: ios/APNsModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(APNsModule, NSObject)

RCT_EXTERN_METHOD(getAPNSToken:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Yoshinobu/APNsModule.swift
import Foundation
import React

@objc(APNsModule)
class APNsModule: NSObject {

  @objc
  func getAPNSToken(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Try to get token from UserDefaults
    if let token = UserDefaults.standard.string(forKey: "APNsDeviceToken") {
      print("📱 APNsModule: Returning token from UserDefaults: \(token)")
      resolve(token)
    } else {
      print("⚠️ APNsModule: No token found in UserDefaults")
      resolve(nil)
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
